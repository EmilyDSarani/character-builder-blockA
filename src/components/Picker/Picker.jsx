import React from 'react'

export default function Picker({
    animal,
    setAnimal,
    shirt,
    setShirt,
    pant,
    setPant,
    shoe,
    setShoe
}) {

    const animalOptions = ['lion', 'monkey', 'cow'];
    const shirtOptions = ['overall', 'hawaiian', 'blue']
    const pantOptions = ['jeans', 'jogger', 'tights']
    const shoeOptions = ['colorful', 'slipper', 'ballet']

    return (
        <div>
            <label>
                Animal:
                <select value={animal} onChange={(e) => setAnimal(e.target.value)}>
                    {animalOptions.map((picture) => (
                        <option key={picture}> {picture}</option>
                    ))}
                </select>
            </label>
            <label>
                Shirt:
                <select value={shirt} onChange={(e) => setShirt(e.target.value)}>
                    {shirtOptions.map((picture) => (
                        <option key={picture}> {picture}</option>
                    ))}
                </select>
            </label>
            <label>
                Pants:
                <select value={pant} onChange={(e) => setPant(e.target.value)}>
                    {pantOptions.map((picture) => (
                        <option key={picture}> {picture}</option>
                    ))}
                </select>
            </label>
            <label>
                Shoes:
                <select value={shoe} onChange={(e) => setShoe(e.target.value)}>
                    {shoeOptions.map((picture) => (
                        <option key={picture}> {picture}</option>
                    ))}
                </select>
            </label>
        </div>
    )
}
