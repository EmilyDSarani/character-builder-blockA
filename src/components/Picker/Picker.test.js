import {screen, render} from '@testing-library/react';
import Picker from './Picker'

it('render the Picker', () => {
    const {container} = render(<Picker />);
    const selectAnimal = screen.getByLabelText(/Animal/);
    
    expect(selectAnimal).toBeInTheDocument();
    expect(container).toMatchSnapshot();
})

