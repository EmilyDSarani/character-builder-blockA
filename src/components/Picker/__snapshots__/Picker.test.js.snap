// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the Picker 1`] = `
<div>
  <div>
    <label>
      Animal:
      <select>
        <option>
           lion
        </option>
        <option>
           monkey
        </option>
        <option>
           cow
        </option>
      </select>
    </label>
    <label>
      Shirt:
      <select>
        <option>
           overall
        </option>
        <option>
           hawaiian
        </option>
        <option>
           blue
        </option>
      </select>
    </label>
    <label>
      Pants:
      <select>
        <option>
           jean
        </option>
        <option>
           jogger
        </option>
        <option>
           tight
        </option>
      </select>
    </label>
    <label>
      Shoes:
      <select>
        <option>
           colorful
        </option>
        <option>
           slipper
        </option>
        <option>
           ballet
        </option>
      </select>
    </label>
    <label>
      What's Your Catchphrase?
      <input
        type="text"
        value=""
      />
    </label>
    <button
      type="button"
    >
      Add Phrase
    </button>
  </div>
</div>
`;
